pkg/daemon/update.go

```diff
@@ -330,13 +330,20 @@ func removePendingDeployment() error {
 	return runRpmOstree("cleanup", "-p")
 }
 
+//nolint:gocyclo
 func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newConfig *mcfgv1.MachineConfig) (retErr error) {
 	// Extract image and add coreos-extensions repo if we have either OS update or package layering to perform
 
 	if dn.nodeWriter != nil {
 		dn.nodeWriter.Eventf(corev1.EventTypeNormal, "OSUpdateStarted", mcDiff.osChangesString())
 	}
 
+	// TODO(jkyros): both paths (layering/non-layering) have different pre/update/post steps, but the events are the same. I hate to
+	// duplicate the eventing code, but I feel like I should split this function into like "applyLegacyOSChanges" and "applyLayeredOSChanges"
+
+	client := NewNodeUpdaterClient()
```

Comment:
Hmm, shouldn't there already be a client that's been initialized somewhere? Would we be able to use that?

Reply:
Yes there is one in the daemon. using it now.